microsoft/testsuites/vm_extensions/waagent.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import uuid
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension, ExtensionNames
+
+
+@TestSuiteMetadata(
+    area="azure",
+    category="functional",
+    description="BVT for VM Agent",
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class WaAgentBvt(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        Runs an extension and verifies it executed on the remote machine.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_vm_agent(self, log: Logger, node: Node) -> None:
+        # Any extension will do, use CustomScript for convenience.
+        # Use the extension to create a unique file on the test machine.
+        test_file = "/tmp/{0}".format(uuid.uuid4())
+        script = "touch {0} && echo Created {0}".format(test_file)
+        settings = {"commandToExecute": script}
+        extension = node.features[AzureExtension]
+        result = extension.create_or_update(
+            name="CustomScript",
+            publisher="Microsoft.Azure.Extensions",
+            type=ExtensionNames.custom_script,
+            type_handler_version="2.0",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+            force_update_tag=test_file,
+        )
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        # Double-check that the file was actually created.
+        message = "File {0} was not created on the test machine".format(test_file)
```

Comment:
Please try f string, instead of .format.

```suggestion
        message = f"File {test_file} was not created on the test machine"
```

Reply:
same on all other strings.

Reply:
thanks, i'll change that. are those guidelines documented somewhere to avoid extra code iterations?